### ISSUE TITLE
Assignment from Scriban script to Dictionary<object, object> throws.

### ISSUE THREAD

**Issue by juhani-honkala** [scriban/scriban#563]
**Question:**  
Should assignment from a script to `Dictionary<object, object>` work? For example, if I have a `Dictionary` like this:

```
Dictionary<object, object> foo = [];
foo["Bar"] = 1;

scriptContext.SetValue("Foo", foo);
```

Should I be able to set that from Scriban using:
```
{{ Foo.Bar = 2 }}
```

In our testing this throws an error: Unexpected exception while accessing target expression: Unable to cast object of type 'System.Int32' to type 'System.Collections.Generic.IDictionary`2[System.Object,System.Object]'.

Replacing the whole object works fine:
```
{{ Foo = { Bar: 2 } }}
```

Also, reading the value works ok

```
{{ Foo.Bar }}
```

**Comment by juhani-honkala** [comment]
Works with the latest version of Scriban. Broken in the newest Nuget release.

**Comment by xoofx** [comment]
Could you post the exact code you are using so that I can replicate it in a unit test?

**Comment by gavin-hay-dispatch** [comment]
I have a reproduction as I've found the same issue:

``` csharp
[Test]
public void TestAssignValToDictionary()
{
    var dict = new Dictionary<string, string>();
    dict["name"] = "bob";
    var model = new ScriptObject();
    model.Add("dict", dict);
    var context = new TemplateContext();
    context.PushGlobal(model);

    var input = "{{dict.location = \"home\"}}";
    var template = Template.Parse(input);
    var results = template.Render(context);

    input = "{{dict[\"location\"] = \"home\"}}";
    template = Template.Parse(input);
    results = template.Render(context);
}
```

It throws an error with the same cast issue. I found the resolution on line 180 in the DictionaryAccessor.cs class.

I have a pr with the changes and a the above test that passes.

**Comment by xoofx** [comment]
Fixed by #574
